### PR TITLE
Pyscript now saves command output during the same period that redirection does

### LINKED ISSUE #659: Consider what output the pyscript is capturing in its StdSim objects
**Question 1**
When we redirect a command, any output in **postparsing** and **cmdfinalization** hooks is not included in the redirected output. However, any output in **precmd** and **postcmd** hooks is redirected.

To be consistent, should an `app()` call in a **pyscript** capture this same output and exclude the **postparsing** and **cmdfinalization** hooks? Right now `app()` captures it all for `CommandResult.stdout`.
 
**Question 2**
Consider the command: `app('help > out.txt')`

What is the expected result?
1. `CommandResult.stdout` contains the output of the **help** command 
2. `CommandResult.stdout` is empty and `CommandResult.stderr` to contain text only if an error occurred redirecting output

Assume both cases do write the output of **help** to **out.txt** if no error occurred.

### PR DESCRIPTION
**Pull request by kmvanbrunt**
Fixes #659

### CODE REVIEW

File: cmd2/cmd2.py [hunk 1]
```diff
@@ -3026,7 +3036,6 @@ def do_py(self, args: argparse.Namespace) -> bool:
         if self._in_py:
             err = "Recursively entering interactive Python consoles is not allowed."
             self.perror(err, traceback_war=False)
-            self._last_result = CommandResult('', err)
```
Comment (tleonhardt) [inline cmd2/cmd2.py]: CommandResult is now an unused import which needs to be removed so flake8 checks will pass

**Comment by tleonhardt** [comment]
I recommend adding something to the **CHANGELOG** in relation to this PR

**Comment by kmvanbrunt** [comment]
Closing until I can make some final changes.

File: tests/pyscript/stdout_capture.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# flake8: noqa F821
+# This script demonstrates when output of a command finalization hook is captured by a pyscript app() call
```
Comment (tleonhardt) [inline tests/pyscript/stdout_capture.py]: Thanks for adding a test

**Pull request merged**